kiwi/storage/subformat/vagrant_base.py

```diff
@@ -135,13 +139,25 @@ def create_image_format(self):
 
         box_img_files = self.create_box_img(self.temp_image_dir)
 
-        metadata_json = os.sep.join([self.temp_image_dir, 'metadata.json'])
+        metadata_json = os.path.join(self.temp_image_dir, 'metadata.json')
         with open(metadata_json, 'w') as meta:
             meta.write(self._create_box_metadata())
 
-        vagrantfile = os.sep.join([self.temp_image_dir, 'Vagrantfile'])
+        vagrantfile = os.path.join(self.temp_image_dir, 'Vagrantfile')
         with open(vagrantfile, 'w') as vagrant:
-            vagrant.write(self._create_box_vagrantconfig())
+            # autogenerate a Vagrantfile:
+            if self.vagrantconfig.get_embedded_vagrantfile() is None:
+                embedded_vagrantfile = self._create_box_vagrantconfig()
+            # user provided a Vagrantfile
+            else:
+                with open(
+                        os.path.join(
+                            self.xml_state.xml_data.description_dir,
+                            self.vagrantconfig.get_embedded_vagrantfile()
+                        ), "r") as users_vagrantfile:
```

Comment:
open for reading is default, you don't need to set that open bits. If you still want to do so please embed it into single quotes, ```'r'```

Reply:
According to `import this`: "Explicit is better than implicit." I'd keep this in as `'r'` 😉